lookout/style/format/benchmarks/top_repos_quality.py

```diff
@@ -0,0 +1,250 @@
+    :param to_commit: Hash of commit.
+    :param bblfsh: Bblfsh address to use.
+    :return: Report.
+    """
+    port = find_port()
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        db = tempfile.NamedTemporaryFile(dir=tmpdirname)
+        fs = tempfile.TemporaryDirectory(dir=tmpdirname)
+        # clone repository
+        git_dir = os.path.join(tmpdirname, get_repo_name(repository))  # location for code
+        cmd = "git clone --single-branch --branch master %s %s" % (repository, git_dir)
+        process = subprocess.Popen(cmd.split())
+        output, error = process.communicate()
+        assert error is None, "Something goes wrong with repository %s " % repository
+        # train model
+        model_path = train_on_repo(git_dir=git_dir, from_commit=from_commit, to_commit=to_commit,
+                                   db=db.name, fs=fs.name, port=port)
+        # quality report
+        analyzer_name = "lookout.style.format.quality_report"
+        config = '{"style.format.analyzer.QualityReportAnalyzer": {"model_path": "%s", ' \
+                 '"bblfsh_address": "%s"}}' % (model_path, bblfsh)
+        with TestAnalyzer(port=port, db=db.name, fs=fs.name, analyzer=analyzer_name) as analyzer:
+            launch_server("push", fr=from_commit, to=to_commit, port=port,
+                          git_dir=git_dir, config_json=config)
+            # find quality report
+            st = [j for j, row in enumerate(analyzer.logs)
+                  if "Classification report:" in row and "QualityReportAnalyzer" in row]
+            assert len(st) == 1
+            st = st[0]
+            quality_report = analyzer.logs[st]
+            # find rule stat report
+            st = [j for j, row in enumerate(analyzer.logs)
+                  if "Rules summary:" in row and "# Model report for" in row]
+            assert len(st) == 1
+            st = st[0]
+            model_report = analyzer.logs[st]
+    return "\n".join(quality_report.split("\n")[1:-1]), "\n".join(model_report.split("\n")[1:-1])
+
+
+def calc_weighted_avg(arr, col, weight_col=3):
+    """Calculate weighted average."""
+    numerator, denominator = 0, 0
+    for row in arr:
+        numerator += row[col] * row[weight_col]
+        denominator += row[weight_col]
+    if denominator == 0:
+        return 1
+    return numerator / denominator
+
+
+def calc_avg(arr, col):
+    """Calculate average."""
+    numerator, denominator = 0, 0
+    for row in arr:
+        numerator += row[col]
+        denominator += 1
+    if denominator == 0:
+        return 1
+    return numerator / denominator
+
+
+def get_presicion_recall_f1_support(report):
+    """Extract avg / total precision, recall, f1 score, support from report."""
+    for line in report.split("\n"):
+        if "weighted avg" in line:
+            _, prec, recall, f1, support = line.split("` | `")
+            prec, recall, f1, support = [float(f.replace("`", "").replace("|", "")) for f in
```

Comment:
This does not seem right: parsing our own data. Can you get the values directly?

Reply:
How about this Markdown trick:

```markdown
# Your report here

Blah-blah-blah.

Tables.

    <details>
    ```json
    {
    "your_data": "here"
    }
    ```
    </details>
```

you would scan to the end of the file, get the JSON block and then deserialize it. Readers will not pay attention because it will be hidden in a spoiler.

Reply:
as I said https://github.com/src-d/style-analyzer/pull/326#discussion_r235990248 - I don't see an easy way to receive comments directly in python, unfortunately.